kubernetes/provider_config.go

```
@@ -0,0 +1,256 @@
+	}
+	if cfg.Host != "" {
+		// Host has to be the full address (scheme://hostname:port). overrides are
+		// processed too late to be taken into account by defaultServerUrlFor(),
+		// so we replicate the URL normalisation here. See
+		// https://github.com/kubernetes/client-go/blob/v12.0.0/rest/url_utils.go#L85-L87
+		hasCA := len(overrides.ClusterInfo.CertificateAuthorityData) != 0
+		hasCert := len(overrides.AuthInfo.ClientCertificateData) != 0
+		defaultTLS := hasCA || hasCert || overrides.ClusterInfo.InsecureSkipTLSVerify
+		host, _, err := restclient.DefaultServerURL(cfg.Host, "", apimachineryschema.GroupVersion{}, defaultTLS)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to parse host: %s", err)
```

golangci-lint (2.12.2)

[error] ST1005: error strings should not be capitalized

(staticcheck)